api/v1alpha1/kvmcluster_types.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+)
+
+// KVMClusterSpec defines the desired state of KVMCluster
+type KVMClusterSpec struct {
+	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+	// +optional
+	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+	// CalicoNetwork defines the network configuration to be used for Calico CNI.
+	// +optional
+	CalicoNetwork CalicoNetwork `json:"calicoNetwork"`
+
+	// DockerNetwork defines the network configuration to be used for the docker daemon.
+	// +optional
+	DockerNetwork DockerNetwork `json:"dockerNetwork"`
+
+	// UserList is a list of users and SSH keys to be added to each workload cluster machine for SSH access.
+	// +optional
+	// +nullable
+	UserList []SSHUser `json:"userList"`
```

Comment:
We need to figure out the plan for SSH access here. This is just copied from the `KVMConfig` for now.

Reply:
While reading stuff I noticed this: https://github.com/kubernetes-sigs/cluster-api/blob/6b5d9ab3f0d59083f3f7329a31741ef60db04a06/bootstrap/kubeadm/api/v1alpha3/kubeadmconfig_types.go#L71. Not quite sure if it's an equivalent yet though.

Reply:
Looks like it has all of the required fields https://github.com/kubernetes-sigs/cluster-api/blob/6b5d9ab3f0d59083f3f7329a31741ef60db04a06/bootstrap/kubeadm/api/v1alpha4/kubeadmconfig_types.go#L221-L265 I'll switch to that type and ask for reviews from other people involved in CAPI.

Reply:
I changed it. Seems better to me.

Reply:
I was thinking more that we can add it to the [`KubeadmConfig`](https://hackmd.io/0TL8_gPVSS6T2O5KZq5q0A?both#KubeadmConfig) cause it looks like the field is already part of the spec.

Reply:
Oh, even better!